### PR TITLE
General (optional) Client by using reqwest (as developed by @tomhoule)

### PR DESCRIPTION
**Pull request by MidasLamb**
I've been trying to get @tomhoule code working, but I was wondering why the client was only available for WASM. This PR introduces a general usable client locked after the default feature "http".

### CODE REVIEW

**Comment by MidasLamb** [comment]
There is still some cleanup to do, but the bulk is done and I think the suggestion this pr is making (including a http client not only for WASM) is ready for discussion

**Comment by MidasLamb** [comment]
@tomhoule, just a friendly reminder, as you said in #327

File: graphql_client/src/http.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            .reqwest_client
+            .post(&self.endpoint)
+            .header("Content-Type", "application/json")
+            .body(serde_json::to_string(&Q::build_query(variables)).unwrap())
+            .send()
+            .await?;
+
+        let text_response = reqwest_response.text().await?;
+
+        dbg!(&text_response);
```
Comment (tomhoule) [inline graphql_client/src/http.rs:59]: Let's delete this line.

File: graphql_client/src/http.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    /// Perform a query.
+    ///
+    // Lint disabled: We can pass by value because it's always an empty struct.
+    #[allow(clippy::needless_pass_by_value)]
+    pub async fn call<Q: GraphQLQuery + 'static>(
+        &self,
+        _query: Q,
+        variables: Q::Variables,
+    ) -> Result<crate::Response<Q::ResponseData>, ClientError> {
+        // TODO: remove the unwrap
```
Comment (tomhoule) [inline graphql_client/src/http.rs:46]: I think this line is not relevant anymore.

**Review by tomhoule — COMMENTED** [review]
I just started reviewing, and I think we have to work a tiny bit more on it, there's a lot of TODOs that need to be addressed before it goes to the main branch — like you noted in your previous messages Thanks for your effort!

**Comment by tomhoule** [comment]
Yes, for the general question: let's have a `reqwest` feature and keep it as thin a wrapper around vanilla reqwest as possible, on all platforms reqwest supports (that should be automatic, without `cfg` flagging on the graphql_client side). I also think we can skip testing it, since it's pretty hard to have a real http test setup and we don't have the time resources to maintain it.

**Comment by tomhoule** [comment]
This feature is implemented in a simpler way in https://github.com/graphql-rust/graphql-client/pull/371

**Pull request closed without merging**